### PR TITLE
Added instruction to use a local repo for dependencies

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation

We want a way to search for dependencies locally and remotely at the same time (precedence locally) so that we can easily develop and debut
JIRA: https://issues.jboss.org/browse/AEROGEAR-2509

## Description

This PR contains a readme that explains how to configure the development environment to use a local repo.
A script to create and update the repo is provided for both unix (bash script) and windows (powershell).

### CODE REVIEW

File: codequality/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+1. Compile the sources of the aerogear-xamarin-sdk as debug
+2. Create 2 environment variables called `SOLUTION_DIR` and `REPO_DIR`. The first one will point to the folder containing the `aerogear-xamarin-sdk` Visual Studio solution, the second one will be the path of the repository.
+If the repository already exists it will be updated, otherwise it will be created.
+`
+export SOLUTION_DIR=$HOME/work/aerogear-xamarin-sdk
+`
+`
+export REPO_DIR=$HOME/nuget
+`
+3. Run the `updaterepo.sh` script
```
Comment (secondsun) [inline codequality/README.md]: updaterepo.sh or createrepo.sh?

Comment (ziccardi) [reply codequality/README.md]: fixed. I forgot to rename the script.

**Comment by secondsun** [comment]
It looks OK, we should make a follow up issue to add a windows version of the update script as well.

**Comment by secondsun** [comment]
I can also confirm this works inside of mingw on Windows :)

File: codequality/README.md [hunk 2]
```diff
@@ -49,14 +49,14 @@ $ nuget
 
 #### Creating the local repo
 
-1. Compile the sources of the aerogear-xamarin-sdk as debug
+1. Compile the sources of the `aerogear-xamarin-sdk` as `debug`
```
Comment (secondsun) [inline codequality/README.md]: Maybe we should say "In Visual Studio" or give the example msbuild command?

File: codequality/updaterepo.ps1 [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+    write-host "Repository successfully created/updated in $REPO_DIR"
+} 
+Catch {
+   write-host "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+   write-host "ERROR: An error has occurred while creating the repository"
+   write-host "Please check that you have successfully compiled the aerogear-xamarin-sdk modules and that you have used the DEBUG profile"
+   write-host "Details:"
+   write-host "$_"
```
Comment (secondsun) [inline codequality/updaterepo.ps1]: Can we get a line number for the failure here?

File: codequality/updaterepo.ps1 [hunk 4]
```diff
@@ -0,0 +1,49 @@
+$ErrorActionPreference = "Stop"
+
+$SOLUTION_DIR="C:\Users\Massi\work\aerogear\aerogear-xamarin-sdk"
+$REPO_DIR="C:\Users\Massi\repo85"
+
+if (!(Test-Path env:SOLUTION_DIR) -and !(Test-Path env:REPO_DIR)) { 
+    write-host "ËRROR!!!"
```
[the remark this answers is not in the document]
Comment (ziccardi) [reply codequality/updaterepo.ps1]: Oops... :D

File: codequality/README.md (hunk 2 above)
Comment (ziccardi) [reply codequality/README.md]: done

File: codequality/updaterepo.ps1 (hunk 3 above)
Comment (ziccardi) [reply codequality/updaterepo.ps1]: done

**Comment by wei-lee** [comment]
@secondsun @wojta are we happy with the approach that @ziccardi has documented here, and remove the example app from the SDK, and use the showcase app instead?

**Comment by wojta** [comment]
I think it's an interesting way to check NuGets before publishing. But I don't think that it's a proper way for the SDK development. It makes sense that app will be in a separate repo and will be imported to the SDK (not the other way around, which is what I did initially). I still think that there should be direct project dependencies, not NuGets. I can fix my script to support replacing the dependencies.

**Comment by secondsun** [comment]
@wei-lee Sure.

**Comment by ziccardi** [comment]
@wojta : the reason behind this PR is that I believe we discussed the 'double and synced` app approach in a meeting (a review or a standup) and we decided that we didn't want that.

**Comment by wei-lee** [comment]
@ziccardi I think this is good information to keep. Let's merge this anyway.

**Pull request merged**